### PR TITLE
Fix IfStatement without braces

### PR DESCRIPTION
**Pull request by danreeves**
Hi,

The issue arises when the inline if is the last line of a block. e.g.

```js
function warn(message) {
    if (DEBUG) console.warn(message);
}
```
becomes

```js
function warn(message) {
    if (DEBUG)
}
```
resulting in `SyntaxError: Unexpected token }`

I also added tests to make sure if blocks with braces still remove the expressions.

This basically tests whether the tree is:

`IfStatement > BlockExpression > ExpressionStatement > CallExpression`
or
`IfStatement > ExpressionStatement > CallExpression`

Let me know if this is a bad idea or unwanted 😁

### CODE REVIEW

**Comment by danreeves** [comment]
I updated this PR over the weekend to include a fix, and updated the description above to explain it

**Comment by emilos** [comment]
I've fallen into a same case while trying to strip the console.error of the `express` lib, would be great to have the fix pulled in :).

**Comment by danreeves** [comment]
@eventualbuddha Sorry for the ping, but any feedback on this?

File: src/index.js [hunk 1]
```diff
@@ -87,7 +89,8 @@ export default function strip ( options = {} ) {
 					else if ( node.type === 'CallExpression' ) {
 						const keypath = flatten( node.callee );
 						if ( keypath && pattern.test( keypath ) ) {
-							if ( parent.type === 'ExpressionStatement' ) {
+							const parentIsntIf = (parentParent && parentParent.type !== 'IfStatement');
```
Comment (eventualbuddha) [inline src/index.js]: This applies to any place where the removed call is the body of a control structure whose body may contain a `BlockStatement`, such as `WhileStatement` etc. I think we should change this to be more conservative and only ever remove the call (as opposed to replacing with `void 0`) in this case: `CallExpression` inside `ExpressionStatement` inside `BlockStatement`. This probably also applies to the `debugger` stuff above.

In addition, we should add tests that cover things like:
* body of a `while` loop
* body of a C-style `for` loop
* body of a `for-of` loop
* body of a `for-in` loop
* body of a `case` statement
* body of an arrow function
* body of a labeled statement
* alternate of an `if` statement (i.e. the `else` part)

That’s all that comes to mind off the top of my head.

**Comment by danreeves** [comment]
closing as resolved by #9

**Pull request closed without merging**